libs/pages/order/src/lib/order-store/order.facade.tsx

```diff
@@ -1,7 +1,150 @@
+import { useParams } from 'react-router';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+
+import {
+  AddDishDto,
+  DishModel,
+  DishPurgatoryModal,
+} from '@pasnik/api/data-transfer';
+
+import * as service from '../order-store/order.service';
+import { PagesOrderProps } from '../pages-order';
 
 export const useOrderFacade = () => {
-  const store = useOrderStore();
+  const { slug } = useParams<PagesOrderProps>();
+  const queryClient = useQueryClient();
+
+  const orderKey = ['fetch', 'order', slug];
+
+  const orderQuery = useQuery(orderKey, () => service.fetchOrder(slug), {
+    enabled: Boolean(slug),
+  });
+
+  const orderDishesKey = ['fetch', 'order', orderQuery?.data?.id, 'dishes'];
+
+  const dishesQuery = useQuery(
+    orderDishesKey,
+    ({ queryKey }) => {
+      const [, , orderId] = queryKey;
+
+      return service.fetchDishes(orderId!);
+    },
+    {
+      refetchInterval: 5000,
+      enabled: Boolean(slug) && Boolean(orderQuery?.data?.id),
+    }
+  );
+
+  const markAsDeliveredMutation = useMutation((orderId: string) =>
+    service.markAsDelivered(orderId)
+  );
+  const markAsOrderedMutation = useMutation((orderId: string) =>
+    service.markAsOrdered(orderId)
+  );
+  const markAsOpenMutation = useMutation((orderId: string) =>
+    service.markAsOpen(orderId)
+  );
+  const markAsClosedMutation = useMutation((orderId: string) =>
+    service.markAsClosed(orderId)
+  );
+
+  const dishUpdate = useMutation(
+    ({
+      orderId,
+      dishId,
+      payload,
+    }: {
+      orderId: string;
+      dishId: number;
+      payload: AddDishDto;
+    }) => service.updateDish(orderId, dishId, payload),
+    {
+      onMutate: async ({ dishId, payload }) => {
+        await queryClient.cancelQueries(orderDishesKey);
+
+        const prevDishes =
+          queryClient.getQueryData<DishModel[]>(orderDishesKey);
+        queryClient.setQueryData<DishModel[]>(orderDishesKey, (prev = []) => {
+          const index = prev?.findIndex((item) => item.id === dishId);
+
+          return [
+            ...prev.slice(0, index),
+            {
+              ...prev[index!],
+              ...payload,
+            },
+            ...prev.slice(index + 1),
+          ];
+        });
+
+        return { prevDishes };
+      },
+      onSettled: () => {
+        queryClient.invalidateQueries(orderDishesKey);
+      },
+    }
+  );
+
+  const dishDelete = useMutation(
+    ({ orderId, dishId }: { orderId: string; dishId: number }) =>
+      service.deleteDish(orderId, dishId),
+    {
+      onMutate: async ({ dishId }) => {
+        await queryClient.cancelQueries(orderDishesKey);
+
+        const prevDishes =
+          queryClient.getQueryData<DishModel[]>(orderDishesKey);
+        queryClient.setQueryData<DishModel[]>(orderDishesKey, (prev = []) => {
+          const index = prev?.findIndex((item) => item.id === dishId);
+
+          return [...prev.slice(0, index), ...prev.slice(index + 1)];
```

Comment:
why not `filter`?

Reply:
it's just to update the item in same place without changing it's index

Reply:
ah yup, fixed